azure-pipelines.yaml

```diff
@@ -0,0 +1,166 @@
+- script: |
+    safeContainerTag=`echo $(containerTag) | sed 's/[^a-zA-Z0-9]/-/g' | tr '[:upper:]' '[:lower:]'`
+    echo '##vso[task.setvariable variable=safeContainerTag]'$safeContainerTag
+  displayName: set safe container tag as lower case container tag with illegal characters removed
+  # extracts merged PR number from git log if a PR merge to master
+- script: |
+    gitLog=`git log --pretty=oneline --abbrev-commit -1`
+    mergedPrNo=`echo $gitLog | sed -n 's/.*(#\([0-9]\+\)).*/\1/p'`
+    echo '##vso[task.setvariable variable=mergedPrNo]'$mergedPrNo
+  condition: and(succeeded(), contains(variables['Build.SourceBranch'], format('refs/heads/{0}', variables['Build.SourceBranchName'])))
+  displayName: extracts merged PR number if merge to branch
+  # create test folder
+- script: |
+    mkdir test-output
+    chmod 777 test-output
+  displayName: create test output directory
+  # build base image
+- task: DockerCompose@0
+  inputs:
+    action: Build services
+    containerregistrytype: Azure Container Registry
+    azureSubscriptionEndpoint: $(azureSubscriptionEndpoint)
+    azureContainerRegistry: $(azureContainerRegistryFull)
+    dockerComposeFile: docker-compose.yaml
+    projectName: $(Build.Repository.Name)
+    qualifyImageNames: true
+    additionalImageTags: $(containerTag)
+  displayName: build base image
+  # run tests
+- task: DockerCompose@0
+  inputs:
+    containerregistrytype: Azure Container Registry
+    azureSubscriptionEndpoint: $(azureSubscriptionEndpoint)
+    azureContainerRegistry: $(azureContainerRegistryFull)
+    dockerComposeFile: docker-compose.yaml
+    additionalDockerComposeFiles: docker-compose.test.yaml
+    projectName: $(Build.Repository.Name)
+    qualifyImageNames: true
+  displayName: run tests
+  # push container to registry
+- task: DockerCompose@0
+  inputs:
+    action: Push services
+    azureSubscriptionEndpoint: $(azureSubscriptionEndpoint)
+    azureContainerRegistry: $(azureContainerRegistryFull)
+    dockerComposeFile: docker-compose.yaml
+    qualifyImageNames: true
+    additionalImageTags: $(containerTag)
+  displayName: Push container to registry
+  # publish test results
+- task: PublishTestResults@2
+  displayName: Publish test results
+  inputs:
+    testResultsFormat: JUnit
+    testResultsFiles: test-output/junit.xml
+    failTaskOnFailedTests: true
+    testRunTitle: Unit Test
+  # deploy via helm
+- task: HelmDeploy@0
+  inputs:
+    azureSubscriptionEndpoint: $(azureSubscriptionEndpoint)
+    azureResourceGroup: $(azureResourceGroup)
+    kubernetesCluster: $(kubernetesCluster)
+    namespace: $(imageName)-$(safeContainerTag)
+    command: upgrade
+    chartType: filepath
+    chartPath: helm
+    releaseName: $(imageName)-$(containerTag)
+    arguments: --set image=$(azureContainerRegistryFull)/$(imageName):$(containerTag) --set container.redeployOnChange=$(Build.BuildId) --set container.imagePullPolicy=Always --atomic
+    install: true
+    waitForExecution: true
```

Comment:
Might also be a good idea to add the `recreate: true` argument.  This will recreate the pods on deployment which doesn't always happen.  Without this argument the PR deployment may not reflect the latest code changes.

Reply:
Added